core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -350,6 +356,13 @@ object SparkEnv extends Logging {
         "levels using the RDD.persist() method instead.")
     }
 
+    val outputCommitCoordinator = mockOutputCommitCoordinator.getOrElse {
+      new OutputCommitCoordinator(conf)
+    }
```

Comment:
Ah, I see how this could be confusing: the OutputCommitCoordinator class serves as both a server and a client interface, similar to MapOutputStatusTracker and a couple of other classes.  I'll see about splitting the client functionality into a separate `OutputCommitCoordinatorClient` class that's initialized everywhere and leave the OutputCommitCoordinator as a driver-only component.